packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -446,5 +452,675 @@ class _CupertinoTextSelectionControls extends TextSelectionControls {
+       super(key: key);
+
+  final List<Widget> children;
+  final bool isArrowPointingDown;
+
+  @override
+  _CupertinoTextSelectionToolbarContentState createState() => _CupertinoTextSelectionToolbarContentState();
+}
+
+class _CupertinoTextSelectionToolbarContentState extends State<_CupertinoTextSelectionToolbarContent> with TickerProviderStateMixin {
+  AnimationController _controller;
+  int _page = 0;
+  int _nextPage;
+
+  void _handleNextPage() {
+    _controller.reverse();
+    _controller.addListener(_listener);
+    setState(() {
+      _nextPage = _page + 1;
+    });
+  }
+
+  void _handlePreviousPage() {
+    _controller.reverse();
+    _controller.addListener(_listener);
+    setState(() {
+      _nextPage = _page - 1;
+    });
+  }
+
+  void _listener() {
+    if (_controller.value != 0.0) {
+      return;
+    }
+
+    setState(() {
+      _page = _nextPage;
+      _nextPage = null;
+    });
+    _controller.forward();
+    _controller.removeListener(_listener);
+  }
+
+  @override
+  void didUpdateWidget(_CupertinoTextSelectionToolbarContent oldWidget) {
+    // If the children are changing, the current page should be reset.
+    if (widget.children != oldWidget.children) {
+      _page = 0;
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  void initState() {
+    _controller = AnimationController(
+      value: 1.0,
+      vsync: this,
+      // This was eyeballed on a physical iOS device running iOS 13.
+      duration: const Duration(milliseconds: 150),
+    );
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final EdgeInsets arrowPadding = widget.isArrowPointingDown
+      ? EdgeInsets.only(bottom: _kToolbarArrowSize.height)
+      : EdgeInsets.only(top: _kToolbarArrowSize.height);
+
+    return DecoratedBox(
+      decoration: const BoxDecoration(color: _kToolbarDividerColor),
+      child: FadeTransition(
+        opacity: _controller,
+        child: _CupertinoTextSelectionToolbarItems(
+          page: _page,
+          backButton: CupertinoButton(
+            borderRadius: null,
+            color: _kToolbarBackgroundColor,
+            minSize: _kToolbarHeight,
+            onPressed: _handlePreviousPage,
+            padding: arrowPadding,
+            pressedOpacity: 0.7,
+            child: const Text('◀', style: _kToolbarButtonFontStyle),
+          ),
+          dividerWidth: 1.0 / MediaQuery.of(context).devicePixelRatio,
+          nextButton: CupertinoButton(
+            borderRadius: null,
+            color: _kToolbarBackgroundColor,
+            minSize: _kToolbarHeight,
+            onPressed: _handleNextPage,
+            padding: arrowPadding,
+            pressedOpacity: 0.7,
+            child: const Text('▶', style: _kToolbarButtonFontStyle),
+          ),
+          nextButtonDisabled: CupertinoButton(
```

Comment:
From my experimentation, the only time it won't show a next button is if there is only one page.  Otherwise there is always a next button, and on the last page it's disabled.

![Screen Shot 2020-04-10 at 11 06 03 AM](https://user-images.githubusercontent.com/389558/79012504-4d557100-7b1b-11ea-86f2-e1eec148b7d3.png)

Reply:
Interesting. I never noticed that. Thanks for the explanation.